site/app/libraries/DateUtils.php

```diff
@@ -77,4 +77,47 @@ public static function validateTimestamp($timestamp) {
         }
         return true;
     }
+
+
+    /**
+     * Asserts that the provided date is a \DateTime object and converts it to one
+     *  if its a string, returning any error in parsing.
+     *
+     * @param $date \DateTime|string A reference to the date object to assert.  Set to null if failed.
+     * @param $time_zone \DateTimeZone The timezone to parse with
+     * @return null|string The error message or null
+     */
+    public static function assertDate(&$date, \DateTimeZone $time_zone) {
+        if (gettype($date) === 'string') {
+            try {
+                $date = new \DateTime($date, $time_zone);
+            } catch (\Exception $e) {
+                $date = null;
+                return 'Invalid Format!';
+            }
+        } else if ($date instanceof \DateTime) { }
+        else {
+            return "Invalid object type!";
+        }
+        return null;
+    }
+    /**
+     * Converts a provided date reference to a string if it isn't already
+     *
+     * @param $date &\DateTime | &string A reference to the date object to convert
+     */
+    public static function convertDateTimeToString(&$date) {
+        if(gettype($date) === 'string') return;
+        $date = self::dateTimeToString($date);
+    }
```

Comment:
Usage of this?

Reply:
Used somewhere that got deleted, will remove